File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/dbhelpers/QueryCommand.java
/**
 * @file
 * @brief
 * @author vitamin.caig@gmail.com
 */

package app.zxtune.fs.dbhelpers;

import java.io.IOException;

public interface QueryCommand {

  boolean isCacheExpired();

  void updateCache() throws IOException;

  boolean queryFromCache();
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/dbhelpers/QueryCommand.kt
/**
 * @file
 * @brief
 * @author vitamin.caig@gmail.com
 */
package app.zxtune.fs.dbhelpers

import java.io.IOException

interface QueryCommand {
    val isCacheExpired: Boolean

    @Throws(IOException::class)
    fun updateCache()

    fun queryFromCache(): Boolean
}
